### DISCUSSION TITLE
blog/shiny-tip-option-where-to-run/

### DISCUSSION THREAD

**Discussion by giscus** [gadenbuie/garrickadenbuie-com#102] [Comments]
# blog/shiny-tip-option-where-to-run/

Quick Shiny Tip: How to use the shiny.launch.browser option to choose where RStudio runs your app.

https://www.garrickadenbuie.com/blog/shiny-tip-option-where-to-run/

**Comment by giocomai** [comment]
This is great, one of the little things that has been annoying me for ages but never quite took the time to deal with.

A minor thing: in the code chunk above you have a small typo: `requiresNamespace`, rather than `requireNamespace`. 

Cheers!

**Reply by gadenbuie** [reply]
Thanks @giocomai, I just fixed the typo! Glad to hear the post was helpful!

**Comment by MarkHanly** [comment]
Solved my problem, thanks for taking the time to post!

**Comment by AlbertRapp** [comment]
Fantastic post! I used this a lot. The only thing that bothered me was that I had to recall the `options()` thing whenever I restart R (which I do a lot). And since the `.rs.incokeShinyPaneViewer` object isn't available at startup I couldn't put the `options()` call into `.Rprofile`. Originially, I solved this by defining a snippt but it turns out that `.rs.invokeShinyPaneViewer` is just a simple function that you can view by calling the name after startup. Thus in my `.Rprofile` I now use

```r
options(
  shiny.launch.browser = function (url) {
    invisible(.Call("rs_shinyviewer", url, getwd(), "pane", NULL, 
                    PACKAGE = "(embedding)"))
  }
)
```

where 

```r
 function (url) {
    invisible(.Call("rs_shinyviewer", url, getwd(), "pane", NULL, 
                    PACKAGE = "(embedding)"))
  }
```

is exactly what you'd see if you were to put  `.rs.invokeShinyPaneViewer` into the console after startup. Just wanted to leave this here in case anyone has been struggling with the same thing :)
